### ISSUE TITLE
Fail to RETAG a image to other "public" project by a user with develop role

### ISSUE THREAD

**Issue by danfengliu** [goharbor/harbor#6453]
Got an unknow error when RETAG a image to other "public" project by a user with develop role, does a public project can be push images by any user?

**Product steps:**
1. Harbor already has a public project with name of library;
2. Create a user by admin, create a project by admin;
3. Add this user to the new project as a develop member, and push a image to this new project;
4. Go to this image, and perform "RETAG" operation, got unknow error.

![image](https://user-images.githubusercontent.com/44018302/49504206-2bf3cd80-f8b4-11e8-9153-16dbfb22fe43.png)

![image](https://user-images.githubusercontent.com/44018302/49504124-0c5ca500-f8b4-11e8-901a-4796da4931e7.png)

**Comment by zhoumeina** [comment]
@cd1989 ， could you please take a look?

**Comment by cd1989** [comment]
Sure

**Comment by danfengliu** [comment]
@cd1989 I modified this issue, because I learned out that this kind of operation should be forbidden, so this issue's key point to is to give user a reasonable information, not an unknown error.

**Comment by zhoumeina** [comment]
@cd1989 , correct. It should be forbidden. I think one more message should be added when user retag  the image successfully. It should also popup a green notification.

**Comment by cd1989** [comment]
According to my understanding, only admin role has write permission to public project. I have fixed based on this.

**Comment by reasonerjt** [comment]
Is it in the RC code? if yes, let's close this. @wy65701436 @cd1989

**Comment by cd1989** [comment]
Yes, we can close
